### PR TITLE
Fix HOOMD issues with GAFF ForceField

### PR DESCRIPTION
**Pull request by CalCraven**
Allow HOOMD writer to write out parameters from a GAFF forcefield.

### CODE REVIEW

File: gmso/external/convert_hoomd.py
Comment (CalCraven) [reply gmso/external/convert_hoomd.py:1443]: This shouldn't be necessary since base_units isn't given a default argument value.

File: gmso/external/convert_hoomd.py
Comment (daico007) [inline gmso/external/convert_hoomd.py:1179]: Is there supposed to be an `else` clause here to raise this `ValueError`?

File: gmso/external/convert_hoomd.py
Comment (daico007) [inline gmso/external/convert_hoomd.py:1115]:
```suggestion
```

File: gmso/utils/schema/ff-gmso.xsd
Comment (daico007) [inline gmso/utils/schema/ff-gmso.xsd:268]: Why is this section removed?

**Review by daico007 — COMMENTED** [review]
Some more comments. Mostly minor, but I have a question on the changes to the schema which I think needed to be clarify before merging.

File: gmso/utils/schema/ff-gmso.xsd
Comment (CalCraven) [reply gmso/utils/schema/ff-gmso.xsd:268]: This section seems to be checking that there are no duplicate names for these potentials. But I don't thing we should be checking the names at all, because then we have like 50 dihedrals that have to be named PeriodicTorsionType-1...50, and if any of them are not labeled this way than it'll fail to load. However, the real key to checking for unique tyeps is checking that no combinations of the types and classes that make up the components of the connection type. I'm not certain why these were implemented this way at all, because in Foyer XML you don't need unique names for each type.

**Comment by daico007** [comment]
Included in #765

File: gmso/utils/schema/ff-gmso.xsd
Comment (daico007) [reply gmso/utils/schema/ff-gmso.xsd:268]: Yeah, I think I am ok with that. Let's comment that section out before we can find a way to validate the uniqueness of these types via atomtype/atomclass

**Review by daico007 — APPROVED** [review]
I think all my issues have been resolved, so this should be good to go once all tests pass.

**Pull request merged**
